### PR TITLE
Fix Apple TV TabBarIOS issue (#15081)

### PR DESCRIPTION
**Pull request by dlowder-salesforce**
**Motivation**

Fix flickering in TabBarIOS on Apple TV... issue #15081 

**Test plan**

Existing CI should still pass.  Issue is resolved when testing the example code in #15081 .

### CODE REVIEW

File: React/Views/RCTTabBar.m [hunk 1]
```diff
@@ -175,6 +175,18 @@ - (void)setItemPositioning:(UITabBarItemPositioning)itemPositioning
 
 #pragma mark - UITabBarControllerDelegate
 
+#if TARGET_OS_TV
+
+- (void)tabBarController:(UITabBarController *)tabBarController didSelectViewController:(nonnull UIViewController *)viewController
```
Comment (dlowder-salesforce) [inline React/Views/RCTTabBar.m]: `didSelectViewController` seems to work fine on iOS too... if desired, we could just use this method for both iOS and tvOS.

**Comment by hramos** [comment]
Removed unrelated Danger warning.

File: React/Views/RCTTabBar.m (hunk 1 above)
Comment (javache) [inline React/Views/RCTTabBar.m]: Yes, I'd prefer that :)

Comment (dlowder-salesforce) [inline React/Views/RCTTabBar.m]: Done

**Comment by javache** [comment]
Hmm, this may cause subtle differences. It seems like before we actually rejected all calls to `shouldSelectViewController`, and then the JS side would actually set the tabbar item it wanted, but in the new model, we would always switch tabs, no longer leaving JS a choice.

**Comment by dlowder-salesforce** [comment]
Wow... so JS could actually select a different tab from the one the user picked?  Interesting!

Since tvOS tab bar works very differently, I'd suggest that we go back to my original commit that just fixes tvOS and leaves iOS behavior unchanged.

**Comment by javache** [comment]
Well the idea is that TabBarItemIOS is a controlled component: all state changes need to come from JS and not from native. I'm not sure why this doesn't work correctly on tvOS.

**Comment by dlowder-salesforce** [comment]
In tvOS testing, I find that the way the tab bar works is very different.  If you use the TV remote to navigate from one tab bar item to the next, the new tab bar item is highlighted BEFORE `shouldSelectViewController:` is called.  So there's no way to avoid changing the state of the tab bar prior to JS being called.  For this reason, I think using `didSelectViewController:` is better for tvOS.

**Comment by javache** [comment]
Ok, makes sense. It does mean it's possible for JS state and native state to go out of sync though. One way classes like RCTTextView solve this is by accepting the native state change, and having the JS side immediately call setNativeProps with the desired state (even if it's the same as the old one).

**Comment by dlowder-salesforce** [comment]
Did additional testing of the issue... this patch is part of the solution, but some additional changes are needed.  Closing this, and will submit the complete fix later.

**Pull request closed without merging**